src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -396,10 +389,14 @@ export class ChallengesubmitComponent implements OnInit {
     const submissionFile = self.globalService.formItemForLabel(self.components, 'input_file').fileValue;
     const submissionProjectUrl = self.globalService.formValueForLabel(self.components, 'project_url');
     const submissionPublicationUrl = self.globalService.formValueForLabel(self.components, 'publication_url');
-    const regex = new RegExp('([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?');
-    if (submissionFile === null || submissionFile === '') {
+    const submissionFileUrl = self.globalService.formItemForLabel(self.components, 'file_url');
+    const regex = new RegExp(/(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-/]))?/);
+    if (self.isSubmissionUsingUrl === false && (submissionFile === null || submissionFile === '')) {
       self.submissionError = 'Please upload file!';
       return;
+    } else if (self.isSubmissionUsingUrl === true && (submissionFileUrl !== '' && !self.validFileUrl)) {
+      self.submissionError = 'Please enter a valid URL which ends in json, zip or csv file extension!';
```

Comment:
This message will confuse the user if the URL is not valid even after entering the url which ends in `zip, json or csv`. Can we just make this simple `Please enter a valid Submission URL`? And as we haven't specified which files are allowed for submission while file uploading also so we'll handle this thing in a separate PR.

Reply:
ok Sir